auditlog/models/rule.py

```diff
@@ -215,41 +215,30 @@ def create(self, vals, **kwargs):
 
     def _make_read(self):
         """Instanciate a read method that log its calls."""
-        # FIXME: read() seems a bit tricky, improve to handle old/new api
-
-        # @api.v7
-        # def read(self, cr, user, ids, fields=None, context=None,
-        #          load='_classic_read', **kwargs):
-        #     print "LOG READ", fields, load, kwargs
-        #     # avoid loops
-        #     if self.env.context.get('auditlog_method_intercepted'):
-        #         return read.origin(
-        #             self, cr, user, ids, fields, context, load, **kwargs)
-        #     # call original method with a modified context
-        #     context = dict(
-        #         self.env.context, auditlog_method_intercepted=True)
-        #     result = read.origin(
-        #         self.with_context(context),
-        #         cr, user, ids, fields, context, load, **kwargs)
-        #     print "RESULT", result
-        #     return result
-
-        # @api.v8
-        # def read(self, fields=None, load='_classic_read', **kwargs):
-        #     print "LOG READ", fields, load, kwargs
-        #     # avoid loops
-        #     if self.env.context.get('auditlog_method_intercepted'):
-        #         return read.origin(self, fields, load, **kwargs)
-        #     # call original method with a modified context
-        #     context = dict(
-        #         self.env.context, auditlog_method_intercepted=True)
-        #     result = read.origin(
-        #         self.with_context(context), fields, load, **kwargs)
-        #     print "RESULT", result
-        #     return result
 
         def read(self, *args, **kwargs):
             result = read.origin(self, *args, **kwargs)
+            # Sometimes the result is not a list but a dictionary
+            try:
```

Comment:
I really don't know. Is there any drawback to test the type directly? We can assume that there are always int or list types passed in parameter.

As performance are important here, I opt for a ``try/except``: lists of IDs are the most common (no check = no performance loss) while an ID alone is the exception, and we fall in the ``except`` in this case. A test implies performance loss on all ``read``, I thought it was not desirable.

I was thinking I could drop some ``isinstance()`` at some places too.